### PR TITLE
Refactor/user view

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: webapp/loggedin/appViewExport.js [hunk 1]
```diff
@@ -15,6 +17,8 @@ import * as SurveyViewsState from './surveyViews/surveyViewsState'
 import * as TableViewsState from './tableViews/tableViewsState'
 import * as JobState from './appJob/appJobState'
 import * as SideBarState from './appSideBar/appSidebarState'
+import * as UserViewState from './modules/users/user/userViewState'
+import * as UserInviteViewState from './modules/users/userInvite/userInviteViewState'
```
Comment (minotogna) [inline webapp/loggedin/appViewExport.js]: can we use name imports? ``'@webpp..```? what do you think?

File: webapp/loggedin/modules/users/components/userGroupDropdown.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
```
Comment (minotogna) [inline webapp/loggedin/modules/users/components/userGroupDropdown.js]: I'd rather call this DropdownUserGroup. what do you think?

File: webapp/loggedin/modules/users/components/userGroupDropdown.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      selection={groups.find(group => AuthGroup.getUuid(group) === selectedGroupUuid)}
+      onChange={group => onChange(AuthGroup.getUuid(group))}
+      readOnlyInput={true}
+    />
+  )
+}
+
+UserGroupDropdown.defaultProps = {
+  editingSelfUser: false, // True if user being edited is the logged one
+  selectedGroupUuid: null,
```
Comment (minotogna) [inline webapp/loggedin/modules/users/components/userGroupDropdown.js]: can this just be groupUuid? or groupUuidSelection?

File: webapp/loggedin/modules/users/components/userGroupDropdown.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      itemLabelFunction={group => i18n.t(`authGroups.${AuthGroup.getName(group)}.label_plural`)}
+      selection={groups.find(group => AuthGroup.getUuid(group) === selectedGroupUuid)}
+      onChange={group => onChange(AuthGroup.getUuid(group))}
+      readOnlyInput={true}
+    />
+  )
+}
+
+UserGroupDropdown.defaultProps = {
+  editingSelfUser: false, // True if user being edited is the logged one
```
Comment (minotogna) [inline webapp/loggedin/modules/users/components/userGroupDropdown.js]: this maybe editingSelf?

File: webapp/loggedin/modules/users/user/actions.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    dispatch(
+      showNotification('userView.removeUserConfirmation', {
+        user: User.getName(userToUpdate),
+        survey: Survey.getLabel(surveyInfo, lang),
+      }),
+    )
+    history.push(appModuleUri(userModules.users))
+  } catch (error) {
+    dispatch(showNotification('appErrors.generic', { text: error }, AppNotificationState.severity.error))
+  }
```
Comment (minotogna) [inline webapp/loggedin/modules/users/user/actions.js]: isn't this covered by the error middleware?

File: webapp/loggedin/modules/users/userInvite/userInviteView.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+.user-invite-view {
```
Comment (minotogna) [inline webapp/loggedin/modules/users/userInvite/userInviteView.scss]: I would call this just ```.user-invite```

File: webapp/loggedin/appViewExport.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/loggedin/appViewExport.js]: ok, it's better, sure

File: webapp/loggedin/modules/users/components/userGroupDropdown.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/components/userGroupDropdown.js]: ok

File: webapp/loggedin/modules/users/components/userGroupDropdown.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/components/userGroupDropdown.js]: ok

File: webapp/loggedin/modules/users/components/userGroupDropdown.js (hunk 4 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/components/userGroupDropdown.js]: I prefer something like editingLoggedUser... editingSelf seems related to the group, not to the user

File: webapp/loggedin/modules/users/user/actions.js (hunk 5 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/user/actions.js]: yes, right!

File: webapp/loggedin/modules/users/userInvite/userInviteView.scss (hunk 6 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/userInvite/userInviteView.scss]: ok

Comment (SteRiccio) [reply webapp/loggedin/modules/users/userInvite/userInviteView.scss]: should we remove the suffix -view even in userView?

File: webapp/loggedin/modules/users/components/dropdownUserGroup.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      itemLabelFunction={group => i18n.t(`authGroups.${AuthGroup.getName(group)}.label_plural`)}
+      selection={groups.find(group => AuthGroup.getUuid(group) === groupUuid)}
+      onChange={group => onChange(AuthGroup.getUuid(group))}
+      readOnlyInput={true}
+    />
+  )
+}
+
+DropdownUserGroup.defaultProps = {
+  editingSelfUser: false, // True if user being edited is the logged one
```
Comment (minotogna) [inline webapp/loggedin/modules/users/components/dropdownUserGroup.js]: in the component you call this editingLoggedUser. is the same property?

File: webapp/loggedin/modules/users/userInvite/userInviteView.scss (hunk 6 above)
Comment (minotogna) [reply webapp/loggedin/modules/users/userInvite/userInviteView.scss]: I think so

File: webapp/loggedin/modules/users/components/dropdownUserGroup.js (hunk 7 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/components/dropdownUserGroup.js]: it SHOULD be the same property... right

**Pull request merged**